### ISSUE TITLE
Allow IPs for Terraform Cloud

### ISSUE THREAD

**Issue by cvillers** [namecheap/terraform-provider-namecheap#62]
Terraform Cloud must be allowed API access. Currently, I have to manually allow each IP address so that applying via Terraform Cloud succeeds. From experimentation, it seems that different IPs are used for different runs, so I have to add a new one after each failure.

A related issue is #59 - if there is a public CIDR for cloud-apply that could also work here. Note that the [IP Range API](https://www.terraform.io/docs/cloud/api/ip-ranges.html) does not return the cloud-apply ranges so I can't predict ahead of time what to use. 

### Terraform Version
1.0.11

### Namecheap provider version
2.0.2

### Expected Behavior
I should be able to use Terraform Cloud as my backend and perform an apply from there.

### Actual Behavior
```
$ terraform apply

╷
│ Error: Invalid request IP: 3.145.98.94 (1011150)
│ 
│   with namecheap_domain_records.MY_DOMAIN,
│   on domain.tf line 3, in resource "namecheap_domain_records" "MY_DOMAIN":
│    3: resource "namecheap_domain_records" "MY_DOMAIN" {
│ 
╵
```

### Steps to Reproduce
1. Use Terraform Cloud (the service, not self-hosted) as the backend.
2. `terraform apply`

**Comment by cvillers** [comment]
After some experimentation, I found that 2 IPs that have failed so far are part of AWS us-east-2, and another is in us-east-1. I understand that it is probably undesirable to allow entire AWS CIDR ranges so #59 may not solve this after all.

**Comment by adamdecaf** [comment]
I don't know the Namecheap internals but their systems might take a few minutes to fully update each allowed IP.

**Comment by StyleT** [comment]
Hi!
Unfortunately this is a limitation of the Namecheap public API itself, rather than Terraform provider itself. Also there is no way to change whitelist via API as far as I know :(

I would advice you to contact our customer support team. They do track such client requests and as soon as there is enough demand on certain feature - it get's processed by respective product team.

**Comment by tedmiston** [comment]
This seems to make the Namecheap provider totally unusable when Terraform Cloud is the backend.  I keep whitelisting new IPs, but from the TFC side it seems that the IPs are dynamic, so it fails every time.

It's pretty hard to do anything real with this provider if the underlying API breaks on requiring whitelisting something that Terraform Cloud doesn't have (static IPs).

**Comment by Lumexralph** [comment]
As @tedmiston said, I wanted to use this provider but am blocked by the requirement to whitelist IP addresses.

- What do you do if you want to run the Terraform locally?
- What do you do if you are using Terraform Cloud?

**Comment by zedtux** [comment]
That's really confusing to have build a Terraform module but having this limitation. Allowing a range of IP addresses should be a good way to workaround this.

**Comment by jmdots** [comment]
Even without a Namecheap API that updates the whitelist, maybe a provider for updating the whitelist could be made to drive the webpage that takes the input. 

In that way, the namecheap provider could depend on the namecheap whitelist provider. When running in Terraform Cloud, a "data" provider could then determine the source IP on the fly and open the door ahead of actual API calls.

Just an idea. This is a major bummer when I just booked a domain with Namecheap for several years. :(

**Comment by benmezger** [comment]
I've added Namecheap provider to our infrastructure but had to completely remove it since we are unable to run this in a CI/CD, which makes this provider basically useless if you don't have a static IP.

**Comment by diegosogari** [comment]
> [!IMPORTANT]
> You can register a Terraform Cloud Agent to run your pipelines. I've tested it and it works!

For people who might still be struggling with this issue, here are some resources that can help you:

- https://developer.hashicorp.com/terraform/cloud-docs/agents
- https://developer.hashicorp.com/terraform/tutorials/cloud/cloud-agents

#59 #67

**Comment by vetal2409** [comment]
Duplicate of #59
